### PR TITLE
Issue/3725 Polish connected/disconnected state UIs

### PR DESCRIPTION
**Pull request by kidinov**
The PR makes changes to the UI of the following screens:

* vaw3DvewbUwuQnPXpdNGGH-fi-16%3A673525
* vaw3DvewbUwuQnPXpdNGGH-fi-16%3A5551

### How to test
Settings -> Manage reader -> Redirect -> Connect to a reader 

### Video/screns

https://user-images.githubusercontent.com/4923871/120320208-99004680-c2ea-11eb-8b39-866b9f255f43.mp4

image
image

image

image

### CODE REVIEW

File: WooCommerce/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -702,8 +702,10 @@
         Card Reader Detail
     -->
     <string name="card_reader_detail_not_connected_header">Connect your card reader</string>
+    <string name="card_reader_detail_learn_more">&lt;a href="https://woocommerce.com/payments/">Learn more&lt;/a> about accepting payments with your mobile device and ordering card readers</string>
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/values/strings.xml:705]: @kidinov I was just poking around, and I noticed the Html tags on this string resource, I think you should confirm with someone from platform9 whether you can use those or not, unless something changed recently, the HTML tags were being removed before sending them for translation: paqN3M-e0-p2

File: WooCommerce/src/main/res/drawable-night/img_card_reader_not_connected.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (malinajirka) [inline WooCommerce/src/main/res/drawable-night/img_card_reader_not_connected.xml:1]: ❓ I haven't verified it's feasible, so take it with a grain of salt -> Have you tried using color references which are different for light vs dark mode instead of introducing a new copy of the drawable? What do you think about this approach?

File: WooCommerce/src/main/res/layout/fragment_card_reader_detail_connected_state.xml [hunk 3]
```diff
@@ -63,4 +69,13 @@
         android:layout_marginHorizontal="@dimen/major_100"
         android:layout_marginTop="@dimen/minor_100"
         tools:text="@string/card_reader_detail_connected_disconnect_reader" />
+
+    <Space
```
Comment (malinajirka) [inline WooCommerce/src/main/res/layout/fragment_card_reader_detail_connected_state.xml:73]: I've never used `space` with weight and 0 width/height - how does this work exactly, especially since this is the only view which has weight set to a non-zero value?

File: WooCommerce/src/main/res/values/strings.xml (hunk 1 above)
Comment (AliSoftware) [reply WooCommerce/src/main/res/values/strings.xml:705]: I think it is indeed a good idea to use HTML entities like `&lt;` and `&gt;` here.

 - We don't do anything to the `strings.xml` files before GlotPress imports them (in fact GlotPress imports a copy of the `strings.xml` file that we freeze during code-freeze, but we still don't modify the file when we copy/freeze it).
 - We _do_ post-process the `strings.xml` files when we download them back from GlotPress, and during that process our XML parser might transform inner tags inside `<string>` tags back into HTML entities indeed, but it won't touch the original one in `values/`.

In any case, as the P2 from Tyler mentions, and as done on other existing strings having HTML tags in them, it's probably indeed a good idea anyway to also transform the `>` into `&gt;` there (even if, as long as the opening bracket is already escaped as `&lt;` that shouldn't lead to any trouble, but better be consistent)

**Review by malinajirka — COMMENTED** [review]
Looks great overall, thank you! 

When we double-check the issue mentioned in [this comment](https://github.com/woocommerce/woocommerce-android/pull/4116#discussion_r643245114), we can proceed with the merge.

File: WooCommerce/src/main/res/values/strings.xml (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/res/values/strings.xml:705]: Thanks @AliSoftware for the input, do you know if something changed recently, because the P2 talks about missing text sent to translators, but from what you say, it seems it shouldn't happen?

@kidinov you may even consider implementing the clickable link in code, to avoid having the underlined text and align with the design, we have a custom span in the code that we use for clickable content: https://github.com/woocommerce/woocommerce-android/blob/develop/WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/WooClickableSpan.kt

Comment (AliSoftware) [reply WooCommerce/src/main/res/values/strings.xml:705]: Well tbh I don't think I ever saw an _unescaped_ opening bracket in the strings slipping through from the time I have been managing Woo releases. So I don't think we ever had an unescaped `<` literal in `values/strings.xml` file since that post? Hence never hitting that issue since.

For example if I search the `values/strings.xml` file for any `<` not appearing at the beginning of a line nor followed by a closing `/string>` or `/item>` – using the following regex: `(?<!\n *)<(?!/(string|item))` –, then besides the opening `<?xml` tag that my regex didn't exclude, the only line of significance that I can find is the `<![CDATA` and `<%2$s` matches from this line:
```xml
    <string name="shipping_label_payments_account_info"><![CDATA[Credit cards are retrieved from the following WordPress.com account: %1$s <%2$s>]]></string>
```
Which uses CDATA to wrap the tag and should thus work as expected at runtime.
And as for the risk of being eaten during the GlotPress import/export process, if I check that string in other languages that got imported after GlotPress translations, e.g. in `values-fr/strings.xml`, it is properly escaped during our post-processing:
```xml
    <string name="shipping_label_payments_account_info">Les cartes de crédit sont récupérées à partir du compte WordPress.com suivant : %1$s &lt;%2$s&gt;</string>
```
So seems we're good there.

Similarly, if I use the regex `(?<!((string|item|plurals) [^>]*"|/string|/item|/plurals|--))>` to try and find `>` that are not preceded by `<string …>` or `<item …>` or `<plurals …>` opening or closing tags nor preceded by `--` (XML comments closing tags `-->`), the remaining matches that are of significance – once I skip over `<resources>` and similar not excluded by the regex – is only that same `shipping_label_payments_account_info` string that uses `CDATA`, which as we've seen ends up having that `>` being escaped as `&gt;` once it went thru GlotPress and back.

---

So TL;DR: all in all, it seems that we don't have any case of `<b>` or similar _unescaped_ HTML tags in the `values/strings.xml` as of today, which means we haven't risked encountering that issue mentioned by Tyler's P2 again – as everybody seems to have escaped their HTML entities properly lately.

File: WooCommerce/src/main/res/layout/fragment_card_reader_detail_connected_state.xml (hunk 3 above)
Comment (kidinov) [reply WooCommerce/src/main/res/layout/fragment_card_reader_detail_connected_state.xml:73]: in LinearLayout, if a view has a weight more than 0 and others views don't have the weight set at all then the first view occupies all the space available. So basically it just pushes other views to the bottom or left/right depends on the orientation and locale.

I think this case is written [there](https://developer.android.com/guide/topics/ui/layout/linear#prioritize-weight)

File: WooCommerce/src/main/res/drawable-night/img_card_reader_not_connected.xml (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/res/drawable-night/img_card_reader_not_connected.xml:1]: IIRC yes it should work, but tbh I wouldn't go this way because currently it's all automated - you take the picture from Figma and android studio converts it to the vector drawable. The chance of error is very small. But with this approach, we need to check if the dark image is the same and create multiple colors for it. IMHO 1-2kb of the size reduction doesn't worth it.

Also, I checked the current case, and the difference between dark/non-dark picture is quite big actually (small piece of the diff):

image

Comment (malinajirka) [reply WooCommerce/src/main/res/drawable-night/img_card_reader_not_connected.xml:1]: 👍 interesting that there are chances even in the paths 😕 . I agree that it's not worth it for this image. 

In general I prefer to use a single file as we already have 99% of the colors defined and usually the 1% is something we don't want to introduce and we can just ping the designer that the colors are not consistent with the rest of the app. But since there are changes even in the paths, I agree that your approach is beter.

File: WooCommerce/src/main/res/layout/fragment_card_reader_detail_connected_state.xml (hunk 3 above)
Comment (malinajirka) [reply WooCommerce/src/main/res/layout/fragment_card_reader_detail_connected_state.xml:73]: Interesting ;)! I've been always avoiding weights, since they used to be super slow on low end devices. But that's not the case anymore. Thanks for the info! I'm curious what will happen when we wrap the screen with a scroll view 🤔.

**Review by malinajirka — APPROVED** [review]

File: WooCommerce/src/main/res/drawable-night/img_card_reader_not_connected.xml (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/res/drawable-night/img_card_reader_not_connected.xml:1]: yeah, It should work just fine for monocolored icons

**Comment by hichamboushaba** [comment]
I see this was merged already, are there plans to address the issue raised [here](https://github.com/woocommerce/woocommerce-android/pull/4116#discussion_r643245114) in a separate PR?

TLDR for possible solutions from the comments:
1. Handle the clickable link in code to use `WooClickableSpan` and remove the underline as it's not matching the design and the rest of the app.
2. Use HTML entities to fix the GlotPress issue, but it will still keep the underline under "Learn more".

**Comment by kidinov** [comment]
@hichamboushaba Probably I got it wrongly, but from my understanding what @AliSoftware said is that it's ok to use escaped HTML in strings?

**Comment by hichamboushaba** [comment]
> @hichamboushaba Probably I got it wrongly, but from my understanding what @AliSoftware said is that it's ok to use escaped HTML in strings?

I think the sign `>` is still not escaped, it should be replaced by `&gt;`.

And sorry to nitpick on this point, but did we confirm with the design team that underlined links are OK in this case? 🙂

**Comment by kidinov** [comment]
> I think the sign > is still not escaped, it should be replaced by &gt;.

🤦  Ill fix it in the next PR

> And sorry to nitpick on this point, but did we confirm with the design team that underlined links are OK in this case? 🙂

Good point. I;; check with design team

**Comment by malinajirka** [comment]
> I think the sign > is still not escaped, it should be replaced by >.

My bad, sorry - I thought it was already fixed 😞 .

> And sorry to nitpick on this point, but did we confirm with the design team that underlined links are OK in this case? 

We plan to do a full design review before this feature gets released - I think we can leave it for that, wdyt @kidinov ?

**Comment by kidinov** [comment]
@malinajirka I asked already and having it underlined is ok! p1622729783206400-slack-C6H8C3G23
